lib/commons/dom/get-element-stack.js

```diff
@@ -0,0 +1,489 @@
+ * References:
+ * https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context
+ * https://github.com/gwwar/z-context/blob/master/devtools/index.js
+ * @param {VirtualNode} vNode
+ * @return {Boolean}
+ */
+function isStackingContext(vNode) {
+	const node = vNode.actualNode;
+
+	//the root element (HTML)
+	if (
+		!node ||
+		node.nodeName === 'HTML' ||
+		node.nodeName === '#document-fragment'
+	) {
+		return true;
+	}
+
+	const computedStyle = vNode.computedStyle;
+
+	// position: fixed or sticky
+	if (
+		vNode.getComputedStylePropertyValue('position') === 'fixed' ||
+		vNode.getComputedStylePropertyValue('position') === 'sticky'
+	) {
+		return true;
+	}
+
+	// positioned (absolutely or relatively) with a z-index value other than "auto",
+	if (
+		computedStyle.zIndex !== 'auto' &&
+		vNode.getComputedStylePropertyValue('position') !== 'static'
+	) {
+		return true;
+	}
+
+	// elements with an opacity value less than 1.
+	if (vNode.getComputedStylePropertyValue('opacity') !== '1') {
+		return true;
+	}
+
+	// elements with a transform value other than "none"
+	const transform =
+		vNode.getComputedStylePropertyValue('-webkit-transform') ||
+		vNode.getComputedStylePropertyValue('-ms-transform') ||
+		vNode.getComputedStylePropertyValue('transform') ||
+		'none';
+
+	if (transform !== 'none') {
+		return true;
+	}
+
+	// elements with a mix-blend-mode value other than "normal"
+	if (
+		vNode.getComputedStylePropertyValue('mix-blend-mode') &&
+		vNode.getComputedStylePropertyValue('mix-blend-mode') !== 'normal'
+	) {
+		return true;
+	}
+
+	// elements with a filter value other than "none"
+	if (
+		vNode.getComputedStylePropertyValue('filter') &&
+		vNode.getComputedStylePropertyValue('filter') !== 'none'
+	) {
+		return true;
+	}
+
+	// elements with a perspective value other than "none"
+	if (
+		vNode.getComputedStylePropertyValue('perspective') &&
+		vNode.getComputedStylePropertyValue('perspective') !== 'none'
+	) {
+		return true;
+	}
+
+	// elements with isolation set to "isolate"
+	if (vNode.getComputedStylePropertyValue('isolation') === 'isolate') {
+		return true;
+	}
+
+	// transform or opacity in will-change even if you don't specify values for these attributes directly
+	if (
+		vNode.getComputedStylePropertyValue('will-change') === 'transform' ||
+		vNode.getComputedStylePropertyValue('will-change') === 'opacity'
+	) {
+		return true;
+	}
+
+	// elements with -webkit-overflow-scrolling set to "touch"
+	if (computedStyle.webkitOverflowScrolling === 'touch') {
+		return true;
+	}
+
+	// a flex item with a z-index value other than "auto", that is the parent element display: flex|inline-flex,
+	if (
+		vNode.getComputedStylePropertyValue('z-index') !== 'auto' &&
+		vNode.parent
+	) {
+		const parent = vNode.parent;
+		if (
+			parent.getComputedStylePropertyValue('display') === 'flex' ||
+			parent.getComputedStylePropertyValue('display') === 'inline-flex'
```

Comment:
I just noticed the latest [CSS3 display spec](https://www.w3.org/TR/css-display-3/#the-display-properties) allows `display: <display-outside> <display-inside>` so `display: inline-flex` is equivalent to `display: inline-flex`. Do we also want to account for a future inside value of `flex`?

Reply:
Probably not for now, but we'll have to keep an eye on it as browsers start adopting it.

Reply:
Decided to support this since I was already updating the code